### PR TITLE
[validation-test] Fix test for OpenCLSDKOverlay

### PR DESCRIPTION
**Pull request by q2r5**
#### What's in this pull request?

The test for OpenCLSDKOverlay was only checking for a compatible GPU and failing if one wasn't found, instead of checking for a CPU as well.
#### Resolved bug number: ([SR-](https://bugs.swift.org/browse/SR-))

---

Before merging this pull request to apple/swift repository:
- [ ] Test pull request on Swift continuous integration.

<details>
  <summary>

Triggering Swift CI</summary>

The swift-ci is triggered by writing a comment on this PR addressed to the GitHub user @swift-ci. Different tests will run depending on the specific comment that you use. The currently available comments are:

**Smoke Testing**

| Platform | Comment |
| --- | --- |
| All supported platforms | @swift-ci Please smoke test |
| OS X platform | @swift-ci Please smoke test OS X platform |
| Linux platform | @swift-ci Please smoke test Linux platform |

 **Validation Testing**

| Platform | Comment |
| --- | --- |
| All supported platforms | @swift-ci Please test |
| OS X platform | @swift-ci Please test OS X platform |
| Linux platform | @swift-ci Please test Linux platform |

Note: Only members of the Apple organization can trigger swift-ci.
</details>

### CODE REVIEW

**Comment by gribozavr** [comment]
@q2r5 Can we always use the CPU group, if that is more reliable and widely available?  We don't care about whether we hit the GPU or CPU, we only care to check that we can call the API.

**Comment by q2r5** [comment]
@gribozavr Seems reasonable to me. I was just keeping in line with the original test which preferred the GPU over CPU. I'll change it to use CL_DEVICE_TYPE_ALL instead so it will just use whichever group it finds first.

**Comment by gribozavr** [comment]
> I'll change it to use CL_DEVICE_TYPE_ALL instead so it will just use whichever group it finds first.

That'd be very nice!

**Comment by q2r5** [comment]
Recommitted and squashed, so it should be ready to go.

File: validation-test/stdlib/OpenCLSDKOverlay.swift [hunk 1]
```diff
@@ -101,12 +101,20 @@ tests.test("clSetKernelArgsListAPPLE") {
   
   // Connect to a compute device
   //
+<<<<<<< HEAD
```
Comment (gribozavr) [inline validation-test/stdlib/OpenCLSDKOverlay.swift]: Did you mean to leave conflict markers in?

**Comment by q2r5** [comment]
> Did you mean to leave conflict markers in?

Whoops, must have been half asleep when I rebased. Should be fixed now.

File: validation-test/stdlib/OpenCLSDKOverlay.swift [hunk 2]
```diff
@@ -101,12 +101,11 @@ tests.test("clSetKernelArgsListAPPLE") {
...
   //
-  var gpu = 1
-  err = clGetDeviceIDs(nil, cl_device_type(gpu != 0 ? CL_DEVICE_TYPE_GPU : CL_DEVICE_TYPE_CPU), 1, &device_id, nil)
+  err = clGetDeviceIDs(nil, cl_device_type(CL_DEVICE_TYPE_ALL), 1, &device_id, nil)
   if (err != CL_SUCCESS)
   {
-    print("Error: Failed to create a device group!")
-    exit(EXIT_FAILURE)
+        print("Error: Failed to create a device group!")
+        exit(EXIT_FAILURE)
```
Comment (gribozavr) [inline validation-test/stdlib/OpenCLSDKOverlay.swift]: Could you revert the indentation change?  We use 2 spaces.

**Comment by gribozavr** [comment]
@swift-ci Please test

**Comment by q2r5** [comment]
Indentation should be fixed now. Sorry about all the delays.

**Comment by gribozavr** [comment]
@q2r5 I'm sorry, I don't see anything but whitespace changes now.

**Comment by q2r5** [comment]
@gribozavr Sorry about that. The commit should now be just the necessary changes with no erroneous modifications to the original whitespace.

**Comment by gribozavr** [comment]
@q2r5 Thank you!

@swift-ci Please test and merge

**Comment by gribozavr** [comment]
@swift-ci Please test and merge

**Pull request closed without merging**